### PR TITLE
BloodDonor: Switched to record template

### PR DESCRIPTION
**Pull request by mintsoft**
Same as before really; just using structured answer:

image

### CODE REVIEW

**Comment by mintsoft** [comment]
Just noticed this: 
image
is the column width adjustable somehow?

**Comment by javathunderman** [comment]
@mintsoft I'm not sure about the column width, but I'm guessing that you can somehow overwrite the CSS...

ping @moollaza

**Comment by moollaza** [comment]
@mintsoft @javathunderman the CSS route is the only method we have for now, perhaps later it could be a property of the record template data, but the CSS route should be OK for now.

File: lib/DDG/Goodie/BloodDonor.pm [hunk 1]
```diff
@@ -36,43 +36,64 @@ sub table_data {
...
+            push(@criticalResults, $donorType . '-');
         }
-
-        $html .= '</table>';
-        return $output, html => $html, heading => "Donors for blood type ".uc($_);
     }
-    return;
+    
+    my $output = '';
+    my $html = "<table class='blooddonor'>";
```
Comment (moollaza) [inline lib/DDG/Goodie/BloodDonor.pm]: looks like we can remove this :)

File: lib/DDG/Goodie/BloodDonor.pm [hunk 2]
```diff
@@ -36,43 +36,64 @@ sub table_data {
...
+
+    sub to_text
+	{
+		my ($data, $keys) = @_;
+		return join "\n", map {"$_: $data->{$_}";} @{$keys};
+	}
+
+    return to_text(\%record_data, \@record_keys), structured_answer => {
+            id => 'BloodDonor',
+            description => 'Returns available donors for a blood type',
```
Comment (moollaza) [inline lib/DDG/Goodie/BloodDonor.pm]: Is there a good place we can send them with a more at link? Perhaps Wikipedia?

Comment (moollaza) [inline lib/DDG/Goodie/BloodDonor.pm]: If so, let's add a `meta: {}` block

**Comment by mintsoft** [comment]
@moollaza all sorted. One this that struck me as odd is that the properties in the meta hash are camelCase, whilst the ones in templates are snake case: (https://duck.co/duckduckhack/goodie_displaying#setting-display-properties-in-a-goodies-perl)

**Comment by moollaza** [comment]
>One thing that struck me as odd is that the properties in the meta hash are camelCase, whilst the ones in templates are snake case:

Hahaha I never noticed that! I guess it shows the API has been changing a bit. I believe we're gravitating towards CamelCase but will check. I'm sure one day the API will have unified casing ;)

**Comment by moollaza** [comment]
LGTM :+1:

File: share/goodie/blood_donor/blood_donor.css [hunk 3]
```diff
@@ -1,20 +1,3 @@
...
-{
-	border-top: 1px solid rgba(155,155,155,.15);
-}
-
-.zci--answer .blooddonor tr:nth-of-type(1)
-{
-	border-top: 0px;
-}
\ No newline at end of file
+.record .record__cell--key {
```
Comment (moollaza) [inline share/goodie/blood_donor/blood_donor.css]: Whoops just noticed this isn't namespaced with `.zci--blood_donor`

**Comment by mintsoft** [comment]
@moollaza Well spotted!

**Comment by jagtalon** [comment]
Excellent. :+1: 

burns

**Pull request merged**